modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/IgniteTableImpl.java

```diff
@@ -303,39 +305,26 @@ public void removeIndex(String idxName) {
     }
 
     private <RowT> BinaryRow updateTuple(RowT row, List<String> updateColList, ExecutionContext<RowT> ectx) {
-        int nonNullVarlenKeyCols = 0;
```

Comment:
Apparently, in this method we were calculating the number of keys and values changed. Based on this numbers, we create rowAssembler. 
But, further, we set a value for every column in the schema. (for (ColumnDescriptor colDesc : columnsOrderedByPhysSchema)) 
In case we changed only part of the columns, we get an internal RowAssembler exception.

This is my understanding at this point. Please pay attention to this part during the review.